editor/src/components/assets.ts

```diff
@@ -40,6 +41,65 @@ export function getAllProjectAssetFiles(
   return allProjectAssets
 }
 
+function getSHA1Checksum(contents: string): string {
+  return new sha1().update(contents).digest('hex')
+}
+
+export function getProjectContentsChecksums(tree: ProjectContentTreeRoot): GithubChecksums {
+  const contents = treeToContents(tree)
+
+  const checksums: GithubChecksums = {}
+  Object.keys(contents).forEach((filename) => {
+    const file = contents[filename]
+    if (isTextFile(file)) {
+      checksums[filename] = getSHA1Checksum(file.fileContents.code)
+    } else if (isAssetFile(file) && file.base64 != undefined) {
+      checksums[filename] = getSHA1Checksum(file.base64)
+    }
+  })
+
+  return checksums
+}
+
+export function deriveGithubFileChanges(
+  projectChecksums: GithubChecksums,
+  githubChecksums: GithubChecksums | null,
+): GithubFileChanges | null {
+  if (githubChecksums == null) {
+    return null
+  }
+
+  const projectFiles = new Set(Object.keys(projectChecksums))
+  const githubFiles = new Set(Object.keys(githubChecksums))
+
+  const untracked: string[] = []
```

Comment:
This is a very minor one, but we have a convention where we use the `let` keyword to signify that a variable is going to be mutated somewhere down the line. In this case it's extremely minor since it's obvious to see that, but our use of `let` makes it even more immediately obvious.